cmd/argocd-server/commands/argocd_server.go

```diff
@@ -123,6 +125,20 @@ func NewCommand() *cobra.Command {
 				baseHRef = rootPath
 			}
 
+			tracerProvider := trace.NewNoopTracerProvider()
```

Comment:
I am not sure I understand the reason why this is a `NoopTracerProvider` instead of a [real TracerProvider](https://pkg.go.dev/go.opentelemetry.io/otel/sdk/trace#NewTracerProvider). Can you please elaborate?

Reply:
There is a flag for collector address. Only if the collector address is not specified will we use the no-op trace provider.

Reply:
Removed this first because now we don't return the tracer provider.